sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -98,33 +98,65 @@ class CryptographyClient(KeyVaultClientBase):
 
     def __init__(self, key, credential, **kwargs):
         # type: (Union[KeyVaultKey, str], TokenCredential, **Any) -> None
+        self._local_only = kwargs.pop("_local_only", False)
 
         if isinstance(key, KeyVaultKey):
             self._key = key
-            self._key_id = parse_key_vault_id(key.id)
+            try:
+                self._key_id = parse_key_vault_id(key.id)
+            except ValueError:
+                if not self._local_only:
+                    raise
+                self._key_id = None
         elif isinstance(key, six.string_types):
             self._key = None
             self._key_id = parse_key_vault_id(key)
             self._keys_get_forbidden = None  # type: Optional[bool]
         else:
             raise ValueError("'key' must be a KeyVaultKey instance or a key ID string including a version")
 
-        if not self._key_id.version:
+        if not (self._local_only or self._key_id.version):
             raise ValueError("'key' must include a version")
 
         self._local_provider = NoLocalCryptography()
         self._initialized = False
 
-        super(CryptographyClient, self).__init__(vault_url=self._key_id.vault_url, credential=credential, **kwargs)
+        vault_url = "vault_url" if not self._key_id else self._key_id.vault_url
+        super(CryptographyClient, self).__init__(vault_url=vault_url, credential=credential, **kwargs)
 
     @property
     def key_id(self):
-        # type: () -> str
+        # type: () -> Optional[str]
         """The full identifier of the client's key.
 
         :rtype: str
         """
-        return self._key_id.source_id
+        if self._key_id:
+            return self._key_id.source_id
+        return None
+
+    @classmethod
+    def from_jwk(cls, jwk):
+        # type: (Union[JsonWebKey, dict]) -> CryptographyClient
+        """Creates a client that can only perform cryptographic operations locally.
+
+        :param jwk: the key's cryptographic material, as a JsonWebKey or dictionary.
+        :type jwk: JsonWebKey or dict
+        :rtype: CryptographyClient
+
+        .. literalinclude:: ../tests/test_examples_crypto.py
+            :start-after: [START from_jwk]
+            :end-before: [END from_jwk]
+            :caption: Create a CryptographyClient from a JsonWebKey
+            :language: python
+            :dedent: 8
+        """
+        if isinstance(jwk, JsonWebKey):
+            key = vars(jwk)
+        else:
+            key = jwk
+        key_id = key.get("kid")
+        return cls(KeyVaultKey(key_id, key), object(), _local_only=True)
```

Comment:
Please forgive me for launching another expedition into design space, but seeing the KeyVaultKey changes necessary for this to work makes me curious about alternatives. For example, what if `_local_only` were `_jwk`, and the client is responsible for initializing itself with the JWK?

Reply:
That would make sense, and it would be great to avoid making changes to existing models. I think there's a question about how clean the abstraction is if we were to switch from `_local_only` to `_jwk` in the cases where we avoid making remote calls, considering it might be reasonable to want a crypto client from a JWK that still uses the service. If you wanted that then you could just create a client with a KeyVaultKey though, so I think it's reasonable to say that a client created with a JWK is inherently local-only

Reply:
After tinkering with this, it might be too much work to have a CryptographyClient's key material as a JWK. The underlying providers would need to account for the key material coming from a JsonWebKey instead of a KeyVaultKey, and while it might be feasible, it's ultimately more work and more error-prone than creating a KeyVaultKey without an `id`